Ip reconcile cron config via ENV (#317)
k8snetworkplumbingwg/whereabouts: doc/extended-configuration.md

You may want to provide a cron expression to configure how frequently the ip-reconciler runs. This is done via the flatfile.

Look for the following parameter `"reconciler_cron_expression"` located in `script/install-cni.sh` and change to your desired schedule.

alternatively you can speficy the `WHEREABOUTS_RECONCILER_CRON` environment variable in your daemonset definition file to provide the cron expresion.

```
You can speficy the `WHEREABOUTS_RECONCILER_CRON` environment variable in your daemonset definition file to override the default cron expression:
```yaml
       env:
        - name: WHEREABOUTS_RECONCILER_CRON
          value: 30 * * * *

> ```

Reviewer: whoops, this is needed!

> Look for the following parameter `"reconciler_cron_expression"` located in `script/install-cni.sh` and change to your desired schedule.
> 
> alternatively you can speficy the `WHEREABOUTS_RECONCILER_CRON` environment variable in your daemonset definition file to provide the cron expresion.
> 
> ```
>        env:
>         - name: WHEREABOUTS_RECONCILER_CRON
>           value: 30 * * * *
> ```

Reviewer: After re-reading this, I would actually even omit the other option (now that you're provided us a neat API to propagate the cron expression).

```suggestion
You can speficy the `WHEREABOUTS_RECONCILER_CRON` environment variable in your daemonset definition file to override the default cron expression:
```yaml
       env:
        - name: WHEREABOUTS_RECONCILER_CRON
          value: 30 * * * *
```
```

Look for the following parameter `"reconciler_cron_expression"` located in `script/install-cni.sh` and change to your desired schedule.

alternatively you can speficy the `WHEREABOUTS_RECONCILER_CRON` environment variable in your daemonset definition file to provide the cron expresion.

```
       env:
        - name: WHEREABOUTS_NAMESPACE
          valueFrom:
            fieldRef:

>               fieldPath: metadata.namespace

Reviewer: I would omit this unrelated variable from the snippet. It is unrelated.

Author: done

This will give you the service name and the port to use, in this case you'll specify it in the configuration in a `service-name:port` format, the default port for etcd clients is `2379`.

> *Note*: It's important to remember that CNI plugins (typically) run directly on the host and not inside pods. This means that if you use the DNS name (which might look something like `example-etcd-cluster-client.default.svc.cluster.local`) for the service (recommended) make sure that you can resolve those hostnames directly from your hosts. You may find some tips regarding that [here](https://blog.heptio.com/configuring-your-linux-host-to-resolve-a-local-kubernetes-clusters-service-urls-a8c7bdb212a7).

Reviewer: why are you changing this line ?
